src-cpp/server42/src/Workers.cpp

```diff
@@ -298,29 +333,44 @@ namespace org::openkilda {
 
                 auto *payload = reinterpret_cast<org::openkilda::Payload *>(udp_layer->getLayerPayload());
 
-                auto now = std::chrono::time_point_cast<std::chrono::nanoseconds>(
-                        std::chrono::high_resolution_clock::now());
-
-                auto duration = now.time_since_epoch();
-                uint64_t seconds = std::chrono::duration_cast<std::chrono::seconds>(duration).count();
-                uint64_t nanoseconds = std::chrono::duration_cast<std::chrono::nanoseconds>(duration).count();
-
                 std::string flow_id(payload->flow_id);
 
                 if (!fake_duration.count(flow_id)) {
-                    fake_duration[flow_id] = base_latency(rd);
+                    fake_duration[flow_id] = base_latency(rd) * cycles_in_1_ms;
                 }
 
-                uint64_t timestamp = (seconds << 32ul) + (nanoseconds & 0xFFFFFFFF);
+                std::shared_ptr<pcpp::MBufRawPacket> mbuf_raw_ptr(new pcpp::MBufRawPacket());
+
+                mbuf_raw_ptr->initFromRawPacket(rx_mbuf[i], device);
+
+                uint64_t latency_time = fake_duration[flow_id];
+                const uint64_t delta_based_direction = 3 * cycles_in_1_ms;
```

Comment:
Maybe it would be better to give some name to magic number 3 here?

Reply:
Latency delta is based on the direction set to 3ms. Three because it gives a good view in opentsdb.